### PR TITLE
API Wrapper: reformat malformed JSON data

### PR DESCRIPTION
**Pull request by BelleNottelling**
Closes issue #654

Verified this change allows the product configuration to be updated & all settings appear to apply correctly

Reformatted JSON looks like this:

```
{
  "CSRFToken": "a1285a5d9d9d5071ef44e371832a3863",
  "product_category_id": "2",
  "status": "disabled",
  "hidden": "0",
  "setup": "after_payment",
  "icon_url": "",
  "title": "Professional Hosting",
  "slug": "professional-hosting",
  "pricing": {
    "type": "recurrent",
    "once": {
      "setup": "0.00",
      "price": "0.00"
    },
    "recurrent": {
      "1W": {
        "setup": "0.00",
        "price": "0.00",
        "enabled": "1"
      },
      "1M": {
        "setup": "0.00",
        "price": "10.00",
        "enabled": "1"
      },
      "3M": {
        "setup": "0.00",
        "price": "0.00",
        "enabled": "1"
      },
      "6M": {
        "setup": "0.00",
        "price": "0.00",
        "enabled": "1"
      },
      "1Y": {
        "setup": "0.00",
        "price": "0.00",
        "enabled": "1"
      },
      "2Y": {
        "setup": "0.00",
        "price": "0.00",
        "enabled": "1"
      },
      "3Y": {
        "setup": "0.00",
        "price": "0.00",
        "enabled": "1"
      }
    }
  },
  "description": "",
  "id": "4"
}
```

### CODE REVIEW

File: src/library/Api/API.js [hunk 1]
```diff
@@ -167,25 +221,25 @@ const API = {
...
+                return response.json();
+            })
+            .then((response) => {
+                // If the response is an error, call the error handler
+                if (response.error) {
+                    if (typeof errorHandler === 'function') {
+                        errorHandler(response.error);
+                    } else {
+                        console.error(`${response.error.message} (Code: ${response.error.code})`);
+                        console.warn("No error handler was specified. The error was logged to the console. Documentation: https://fossbilling.org/docs/api/javascript   ");
```
Comment (evrifaessa) [inline src/library/Api/API.js:234]: While you’re at it, can you remove the blank space here?

**Review by evrifaessa — APPROVED** [review]
Haven’t tested, but looks like a great PR to me. Thanks for the quick patch.

File: src/library/Api/API.js (hunk 1 above)
Comment (jaapmarcus) [reply src/library/Api/API.js]: Changed it

**Comment by jaapmarcus** [comment]
Works fine.

We should include a version as in 
`<script src="build/js/fossbilling-bundle.min.js?v=0.0.1"></script>`

As it currently looks like 
`<script src="https://xxxx/library/Api/API.js"></script>`


If you domain settings cache is set very strict it might not get updated properly!

**Review by jaapmarcus — CHANGES_REQUESTED** [review]
See my last comment some very small changes....

**Comment by BelleNottelling** [comment]
> See my last comment some very small changes....

Done, now the src is like this:
`script src="https://example.com/library/Api/API.js?v=0.2.6"></script>`

**Review by jaapmarcus — APPROVED** [review]

**Pull request merged**